### PR TITLE
Release/0.7.1

### LINKED ISSUE #126: Update `/permissions/edit/<id>` to allow changes by cataloging admins
A cataloging admin for collection *X* and *Y* should be able to:
* [ ] change `registrant` and `cataloger` settings a permission for X or Y
* [ ] change which user that's referenced by a permission on X/Y
* [ ] change target collection for a permission from X to Y or Y to X

### LINKED ISSUE #129: Mailto: on admin emails
Now users seeking permissions need to copy paste the email, make link pls

### LINKED ISSUE #130: Display admin powers
Information about your admin powers on start page:

* Can create users
* Can assign permissions

### LINKED ISSUE #134: Rubrikändring
Från: "Behörigheter på aktiva sigel" till "Mina behörigheter"

skarmavbild 2017-12-15 kl 10 54 12

### PR DESCRIPTION
**Pull request by mblomdahl**
* Revised API endpoints for registering/editing permissions; now allowing cataloging admins to register new and edit existing permissions on their collections (#126)
* UX enhancements (#129, #134, #130)

Fixes #129 and fixes #126 and fixes #134 and fixes #130.

### CODE REVIEW

File: xl_auth/user/models.py [hunk 1]
```diff
@@ -153,12 +153,12 @@ def is_cataloging_admin(self):
...
-        for permission in self.permissions:
-            if permission.cataloging_admin and permission.collection == collection:
-                return True
-        return False
+    def is_cataloging_admin_for(self, *collections):
+        """Check 'cataloging_admin' status for one or more specific collections."""
+        collections = set(collections)
+        admin_permissions = set(_ for _ in self.permissions
+                                if _.cataloging_admin and _.collection in collections)
+        return bool(len(admin_permissions) and len(admin_permissions) == len(collections))
```
Comment (mxtthias) [inline xl_auth/user/models.py]: What do you think about something like this instead?

```python
def is_cataloging_admin_for(self, *collections):
    collections = set(collections)
    # Can't be cataloging admin for nothing
    if not collections:
        return False
    # Get all collections where this user is cataloging admin
    admin_permission_collections = set(_.collection for _ in self.permissions if _.cataloging_admin)
    # Return True iff the user is cataloging admin for all collections in the input
    return collections.issubset(admin_permission_collections)
```

File: xl_auth/user/views.py [hunk 2]
```diff
@@ -57,26 +57,25 @@ def profile():
 @login_required
 def register():
     """Register new user."""
-    if not current_user.is_admin:
```
Comment (mxtthias) [inline xl_auth/user/views.py]: Same as above.

File: xl_auth/translations/sv/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -583,60 +593,68 @@ msgid "Edit Existing Collection"
...
+#: xl_auth/templates/collections/home.html:6 xl_auth/templates/collections/home.html:71
+#: xl_auth/templates/collections/home.html:136 xl_auth/templates/collections/home.html:191
 msgid "Go to"
 msgstr "Gå till"
 
-#: xl_auth/templates/collections/home.html:6 xl_auth/templates/collections/home.html:66
+#: xl_auth/templates/collections/home.html:9 xl_auth/templates/collections/home.html:86
+#: xl_auth/templates/collections/home.html:143
+msgid "Active Collections without Users"
+msgstr "Aktiva sigler som ej har användare associerade"
```
[the remark this answers is not in the document]
Comment (mblomdahl) [reply xl_auth/translations/sv/LC_MESSAGES/messages.po]: Från @anjabergman :
> Aktiva sigler med en eller flera användare

> Aktiva sigler utan användare

File: tests/forms/test_permission_edit.py [hunk 4]
```diff
@@ -66,39 +76,94 @@ def test_validate_user_id_does_not_exist(superuser, permission):
...
+    other_collection = CollectionFactory().save_as(superuser)
+    PermissionFactory(user=user, collection=other_collection,
+                      cataloging_admin=True).save_as(superuser)
+    other_user = UserFactory().save_as(user)
+    assert user.is_cataloging_admin_for(permission.collection) is True
+    form = EditForm(user, permission.id,
+                    permission_id=permission.id,
+                    user_id=other_user.id,
+                    collection_id=other_collection.id,
+                    cataloging_admin=permission.cataloging_admin,
```
[the remark this answers is not in the document]
Comment (mblomdahl) [reply tests/forms/test_permission_edit.py]: No, the idea is to preserve the original value, true or false. So you cannot make another user cataloging admin unless you're a sysadmin. But you could revoke another's cataloging admin rights either by this form or deleting the permission entirely. Added an inline comment for clarification!

**Pull request merged**